api/grpcserver/grpcserver_test.go

```diff
@@ -477,11 +473,12 @@ func marshalProto(t *testing.T, msg proto.Message) string {
 	return buf.String()
 }
 
-var cfg = DefaultTestConfig()
+func launchServer(tb testing.TB, services ...ServiceAPI) (Config, func()) {
+	cfg := DefaultTestConfig()
 
-func launchServer(tb testing.TB, cfg Config, services ...ServiceAPI) func() {
-	grpcService := New(cfg.PublicListener, logtest.New(tb).Named("grpc"))
-	jsonService := NewJSONHTTPServer(cfg.JSONListener, logtest.New(tb).WithName("grpc.JSON"))
+	// run on random ports
+	grpcService := New("127.0.0.1:0", logtest.New(tb).Named("grpc"))
+	jsonService := NewJSONHTTPServer("127.0.0.1:0", logtest.New(tb).WithName("grpc.JSON"))
```

Comment:
How about updating `DefaultTestConfig()`  to use `127.0.0.1:0` as the listener addresses?

Reply:
I'm a bit nervous as to what might break with this change. It seems that it should only affect tests in the`grpcserver` and `node` packages. I have re-run those and they passed, so it might be OK.